DOL.WHD.Section14c.Log/Repositories/ErrorLogRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using DOL.WHD.Section14c.Log.DataAccess;
+using DOL.WHD.Section14c.Log.Models;
+using NLog;
+using System;
+using System.Linq;
+
+namespace DOL.WHD.Section14c.Log.DataAccess.Repositories
+{
+    public class ErrorLogRepository: IErrorLogRepository
+    {
+        private readonly ApplicationLogContext _dbContext;
+        private static readonly Logger _logger = LogManager.GetCurrentClassLogger();
+        private Boolean Disposed;
+
+        public ErrorLogRepository()
+        {
+            _dbContext = new ApplicationLogContext();
+        }
+
+        /// <summary>
+        /// Get All Logs
+        /// </summary>
+        /// <returns></returns>
+        public IQueryable<APIErrorLogs> GetAllLogs()
+        {
+            return _dbContext.ErrorLogs.AsQueryable();
+        }
+
+        /// <summary>
+        /// Add New Log Message
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public LogDetails AddLog(LogDetails entity)
+        {
+            
+            if (entity != null)
+            {
+                LogEventInfo eventInfo = new LogEventInfo();
+                eventInfo.Properties[Constants.CorrelationId] = Guid.NewGuid().ToString();
+
+                eventInfo.Properties[Constants.EIN] = string.IsNullOrEmpty(entity.EIN) ? string.Empty : entity.EIN;
+                eventInfo.LoggerName = "NLog";
+                if (string.IsNullOrEmpty(entity.Message))
+                {
+                    throw new ArgumentException("Message cannot be null or empty string", "Log Message");
+                }
+
+                eventInfo.Message = entity.Message;
+
+                if (!string.IsNullOrEmpty(entity.Exception))
+                {
+                    eventInfo.Exception = new Exception(entity.Exception);
```

Comment:
I'm curious if this `Exception` object will have a misleading stack trace.  I'm not sure how that's captured in the logs so it may not matter, and anyway, I think the worst case is that it would have one extra call on it.  Shouldn't be too hard to figure out.  Just curious how it'll work out in practice.